src/js/components/ServiceDetailDebugTab.js

```diff
@@ -0,0 +1,97 @@
+import moment from 'moment';
+import React from 'react';
+
+import DescriptionList from './DescriptionList';
+import Service from '../structs/Service';
+
+class ServiceDetailDebugTab extends React.Component {
+  getValueText(value) {
+    if (value == null || value === '') {
+      return (
+        <p>Unspecified</p>
+      );
+    }
+
+    return (
+      <span>{value}</span>
+    );
+  }
+
+  getLastTaskFailureInfo() {
+    let {lastTaskFailure} = this.props.service;
+    if (lastTaskFailure == null) {
+      return (
+        <p>This app does not have failed tasks</p>
+      );
+    }
+
+    const {version, timestamp, taskId, state, message, host} = lastTaskFailure;
+    let timeStampText = 'Just now';
+    if (new Date(timestamp) > Date.now()) {
+      timeStampText = new moment(timestamp).fromNow();
+    }
+    let taskFailureValueMapping = {
+      'Task id': this.getValueText(taskId),
+      'State': this.getValueText(state),
+      'Message': this.getValueText(message),
+      'Host': this.getValueText(host),
+      'Timestamp': <span>{timestamp} ({timeStampText})</span>,
+      'Version': <span>{version} ({new moment(version).fromNow()})</span>
```

Comment:
Also, we probably want to use the `DateUtil` for this instead of using `moment` directly in the application, to be consistent with other components.